### PR TITLE
doc(backing image) : add backing image encryption, backup and store flow via UI

### PR DESCRIPTION
**Pull request by a110605**
#### Which issue(s) this PR fixes:
Add UI flow how to do backing image encrypt, decrypt, backup and restore.

Issue #
[[UI][FEATURE] Add backing image encryption and clone support](https://github.com/longhorn/longhorn/issues/8789)
[[FEATURE] Add BackupBackingImage UI](https://github.com/longhorn/longhorn/issues/7541)

### CODE REVIEW

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md [hunk 1]
```diff
@@ -75,8 +83,23 @@ spec:
...
+Restore backing image from a backup via UI
+1. Go to `Setting -> Backing Image` page.
+2. Click the `Restore` action in operation menu for the backup backing image you want to restore
+3. Input the `Secret` and `Secret Namespace` if you want to restore an encrypted backup backing image
+
+{{< figure src="/img/screenshots/backing-image/restore.png" >}}
+
+> **IMPORTANT:**
+>
+> Longhorn won't store secret / secret namespace in the backup backing image config in v1.7.0 release.
```
Comment (a110605) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md:95]: This is current limitation when restore an encrypted backing image.
Will improve in this ticket in the future : [[IMPROVEMENT] backup backing image should store the secret and secret namespace if it is encrypted](https://github.com/longhorn/longhorn/issues/8884)

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md [hunk 2]
```diff
@@ -75,8 +83,24 @@ spec:
 >   - `concurrent-limit`: Maximum number of worker threads that can concurrently run for each restore operation. When unspecified, Longhorn uses the default value.
 > - `checksum`: You can specify the expected SHA-512 checksum of the backing image file, which Longhorn uses to validate the restored file. When unspecified, Longhorn uses the checksum of the restored file as the truth.
 
+Restore backing image from a backup via UI
+1. Go to `Setting -> Backing Image` page.
+2. Click the `Restore` action in operation menu for the backup backing image you want to restore
+3. Input the `Secret` and `Secret Namespace` if you want to restore an encrypted backup backing image
```
Comment (ChanYiLin) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md:89]: I think this is more accurate.
3. Input the `Secret` and `Secret Namespace` if the backing image is an encrypted image

Comment (ChanYiLin) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md:89]: I think this is more accurate.

`3. Input the Secret and Secret Namespace if the backing image is an encrypted image`

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md [hunk 3]
```diff
@@ -75,8 +83,24 @@ spec:
...
+4. Click OK
+
+{{< figure src="/img/screenshots/backing-image/restore.png" >}}
+
+> **IMPORTANT:**
+>
+> Longhorn won't store secret / secret namespace in the backup backing image config in v1.7.0 release.
+>
+> Users need to manually fill in the secret / secret namespace when restoring the backing image.
```
Comment (ChanYiLin) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-backup.md:98]: `Users need to manually fill in the secret / secret namespace of the image when restoring`

**Review by ChanYiLin — DISMISSED** [review]
LGTM

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md [hunk 4]
```diff
@@ -42,6 +42,13 @@ spec:
 > - `backing-image`: Specify the name of the backing image to be cloned.
 > - `encryption`: Set the value to `ignore` to directly clone the backing image. If the value is not given, Longhorn use `ignore` as default value.
 
+Longhorn also supports cloning the backing image via UI.
+1. Go to `Setting -> Backing Image` page and click `Create Backing Image` button.
+2. Select `Created From` to `Clone From Existing Backing Image` and select `Encryption` to `ignore`.
+3. Click `OK`
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md:48]:
```suggestion
You can also clone a backing image using the Longhorn UI.

1. Go to **Setting** > **Backing Image**.

1. Click **Create Backing Image**.

1. Configure the following settings:

  - **Created From**: Select **Clone From Existing Backing Image**.
  - **Encryption**: Select **Do not encrypt**.

1. Click **OK**.
```

Comment (jillian-maroket) [reply content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md:48]: Given that the other values are `Encrypt` and `Decrypt`, `Do not encrypt`  is more appropriate than `ignore`.

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md [hunk 5]
```diff
@@ -100,8 +107,16 @@ stringData:
 > - `secret`: Specify the secret used to encrypt the backing image.
 > - `secret-namespace`: Specify the namespace of the secret used to encrypt the backing image.
 
+You can also clone and encrypt the backing image via UI
+1. Go to `Setting -> Backing Image` page and click `Create Backing Image` button.
+2. Select `Created From` to `Clone From Existing Backing Image` and select `Encryption` to `encrypt`.
+3. Input the `Secret` and `Secret Namespace` which are required to encrypt the backing image
+4. Click `OK`
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md:114]:
```suggestion
You can also create an encrypted copy of a backing image using the Longhorn UI.

1. Go to **Setting** > **Backing Image**.

1. Click **Create Backing Image**.

1. Configure the following settings:

  - **Created From**: Select **Clone From Existing Backing Image**.
  - **Encryption**: Select **Encrypt**.

1. Specify the secret and secret namespace to be used for encryption.

1. Click **OK**.
```

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md [hunk 6]
```diff
@@ -160,9 +175,17 @@ spec:
 > - `secret`: Specify the secret used to decrypt the backing image.
 > - `secret-namespace`: Specify the namespace of the secret used to decrypt the backing image.
 
+You can also clone and decrypt the backing image via UI
+1. Go to `Setting -> Backing Image` page and click `Create Backing Image` button.
+2. Select `Created From` to `Clone From Existing Backing Image` and select `Encryption` to `decrypt`.
+3. Input the `Secret` and `Secret Namespace` which are required to decrypt the backing image
+4. Click `OK`
```
Comment (jillian-maroket) [inline content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md:182]:
```suggestion
You can also decrypt a backing image (through cloning) using the Longhorn UI.

1. Go to **Setting** > **Backing Image**.

1. Click **Create Backing Image**.

1. Configure the following settings:

  - **Created From**: Select **Clone From Existing Backing Image**.
  - **Encryption**: Select **Decrypt**.

1. Specify the secret and secret namespace to be used for decryption.

1. Click **OK**.
```

**Review by jillian-maroket — COMMENTED** [review]
I reviewed both the doc updates and the UI wording. If you want to create a separate PR for the UI wording updates, I can review that as well.

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md (hunk 4 above)
Comment (a110605) [reply content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md:48]: But `ignore` is the common word in longhorn settings to inherit from global settings or as default option. You can see lots of settings having ignore in their options. 

I won't update the wording at this moment, but we can review the option wordings in [LEP](https://github.com/longhorn/longhorn/blob/master/enhancements/20240612-backing-image-encryption-and-clone-supprt.md#backing-image-data-source) reivew stage.

**Comment by a110605** [comment]
Let me summarize UI suggestion wordings, I can have PR to change it and merge after RC1 build.
| Where to change | Old Wording | New Wording |
| --- | --- | --- |
| Restore Backing image modal title | Restore ${name} Backup Backing Image | Restore Backing Image | 
| New field in restore Backing image modal  | N/A | Backup: ${name} |
| Action in operation menu | Backup | Back Up |
| Table Name | Backup Backing Image | Backing Image Backup | 

cc @ChanYiLin , @derekbit

File: content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md (hunk 4 above)
Comment (ChanYiLin) [reply content/docs/1.7.0/advanced-resources/backing-image/backing-image-encryption.md:48]: `encrypt`, `decrypt`, `ignore` is the extra operation when doing cloning 
this is not only shown on the UI, but will be used in the system as well
So this needs to be a single and simple word.

you can check our settings, we use lots of `ignore` as default value, ref: https://longhorn.io/docs/1.6.2/references/settings/
maybe in the future we can come up with a more appropriate and common word for this case in all the settings.

**Comment by a110605** [comment]
Waiting for change wording PR merged. I will take new screenshot and rebase this PR.

**Comment by a110605** [comment]
Hi @jillian-maroket, @derekbit , @ChanYiLin ,
I've updated the screenshot and wordings. Please have a look this PR again. thanks.

**Review by jillian-maroket — APPROVED** [review]

**Pull request merged**